libs/execution_context/ebpf_maps.c

```diff
@@ -861,15 +861,40 @@ ebpf_map_create(
     return result;
 }
 
-uint8_t*
-ebpf_map_find_entry(_In_ ebpf_map_t* map, _In_ const uint8_t* key, int flags)
-{
+ebpf_result_t
+ebpf_map_find_entry(
+    _In_ ebpf_map_t* map,
+    size_t key_size,
+    _In_reads_(key_size) const uint8_t* key,
+    size_t value_size,
+    _Out_writes_(value_size) uint8_t* value,
+    int flags)
+{
+    uint8_t* return_value;
+    if (!(flags & EBPF_MAP_FLAG_HELPER) && (key_size != map->ebpf_map_definition.key_size)) {
```

Comment:
I think this should be true for non helper calls too, isnt it?

Reply:
No, because non-helper calls originate from untrusted user space.

Reply:
and for UM API too, the value size must be >= map_def.value_size.

Reply:
Why arent you enforcing that the key_size should be definition.key_size irrespective of whether the caller is ebpf program or UM app?

Reply:
For the eBPF program, the key size is checked by the verifier. The eBPF helper function passes just the pointer to the key, so at this layer, we get key size == 0.